pinot-core/src/main/java/org/apache/pinot/core/util/trace/TraceContext.java

```diff
@@ -67,7 +68,7 @@ JsonNode toJson() {
 
     final String _traceId;
     final List<LogEntry> _logs = new ArrayList<>();
-    int _numChildren = 0;
+    AtomicInteger _numChildren = new AtomicInteger(0);
```

Comment:
It can be useful to save memory and indirections with `Atomic*FieldUpdater`s, you can use one as follows:

```java
private static final AtomicIntegerFieldUpdater<Trace> UPDATER = AtomicIntegerFieldUpdater.newUpdater(Trace.class, "_numChildren");
private volatile int _numChildren;
...
return _traceId + "_" + UPDATER.getAndIncrement(this);
```

At the very least, please make `_numChildren` final if it remains an `AtomicInteger`

Reply:
made it final